### PR TITLE
GEODE-5396: A bit of restructuring and a bit of cleanup to isolate fa…

### PR DESCRIPTION
**Pull request by mhansonp**
…ilures better

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by mhansonp** [comment]
It looks like concourse if having problems downloading stuff for gradle. The code "ran on my machine".

File: geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/Server.java [hunk 1]
```diff
@@ -103,26 +160,63 @@ public Server(int jmxPort, String properties, String jsonAuthFile) throws Except
...
+        removeMemberMBean(m);
+      } catch (InstanceAlreadyExistsException e) {
+        e.printStackTrace();
+      } catch (MBeanRegistrationException e) {
+        e.printStackTrace();
+      } catch (NotCompliantMBeanException e) {
+        e.printStackTrace();
+      } catch (MalformedObjectNameException e) {
+        e.printStackTrace();
+      } catch (NullPointerException e) {
```
Comment (mcmellawatt) [inline geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/Server.java]: Do we really want to swallow NullPointerExceptions and just print them?  Seems like at very least we should wrap this and the above exceptions in a TestException so it fails the test.  Or just add them to the exceptions thrown by this method.

File: geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/Server.java [hunk 2]
```diff
@@ -103,26 +160,63 @@ public Server(int jmxPort, String properties, String jsonAuthFile) throws Except
...
+    String[] regions = getArrayProperty(props, "regions");
+    for (String reg : regions) {
+      try {
+        removeRegionMBean(reg);
+      } catch (MBeanRegistrationException e) {
+        e.printStackTrace();
+      } catch (MalformedObjectNameException e) {
+        e.printStackTrace();
+      } catch (NullPointerException e) {
+        e.printStackTrace();
```
Comment (mcmellawatt) [inline geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/Server.java]: Same comment here as above

File: geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/Server.java [hunk 3]
```diff
@@ -194,6 +288,14 @@ private void addMemberMBean(String m) throws InstanceAlreadyExistsException,
     mbs.registerMBean(m1, new ObjectName(Member.OBJECT_NAME + ",member=" + m));
   }
 
+
+  private void removeMemberMBean(String m)
+      throws InstanceAlreadyExistsException, InstanceNotFoundException,
+      MBeanRegistrationException, NotCompliantMBeanException, MalformedObjectNameException {
+    Member m1 = new Member(m);
```
Comment (mcmellawatt) [inline geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/Server.java]: Local variable m1 is unused?

File: geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/rules/ServerRule.java [hunk 4]
```diff
@@ -72,6 +62,41 @@ protected void after() {
     }
   }
 
+  private void startServer() {
+    try {
+      server = null;
```
Comment (mcmellawatt) [inline geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/rules/ServerRule.java]: No need to set server to null here.

File: geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/rules/ServerRule.java [hunk 5]
```diff
@@ -72,6 +62,41 @@ protected void after() {
...
+      ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+      String jmxPropertiesFile = classLoader.getResource("test.properties").getPath();
+      int jmxPort = AvailablePort.getRandomAvailablePort(AvailablePort.SOCKET);
+      System.setProperty(PulseConstants.SYSTEM_PROPERTY_PULSE_PORT, Integer.toString(jmxPort));
+      server = Server.createServer(jmxPort, jmxPropertiesFile, jsonAuthFile);
+      server.start();
+    } catch (Exception e) {
+      throw new Error(e);
```
Comment (mcmellawatt) [inline geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/rules/ServerRule.java]: Is this "Error" type handled specially somewhere?  Why do we wrap all exceptions in an Error?

File: geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/rules/ServerRule.java [hunk 6]
```diff
@@ -72,6 +62,41 @@ protected void after() {
...
+      jetty = JettyHelper.initJetty(LOCALHOST, httpPort, new SSLConfig());
+      JettyHelper.addWebApplication(jetty, PULSE_CONTEXT, getPulseWarPath(), null, null);
+      pulseURL = "http://" + LOCALHOST + ":" + httpPort + PULSE_CONTEXT;
+      System.out.println("Pulse started at " + pulseURL);
+
+      if (jetty != null) {
+        jetty.start();
+      }
+    } catch (Exception e) {
+      throw new Error(e);
```
Comment (mcmellawatt) [inline geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/rules/ServerRule.java]: Same comment here

File: geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/rules/ServerRule.java [hunk 7]
```diff
@@ -72,6 +62,41 @@ protected void after() {
...
+    }
+  }
+
+  // Until we straighten out the Server Code properly, this should be commented out.
```
Comment (mcmellawatt) [inline geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/rules/ServerRule.java]: Not sure I understand this comment.  What should be commented out?  I don't see anything commented out

File: geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/rules/WebDriverRule.java [hunk 8]
```diff
@@ -78,14 +93,17 @@ private void login() {
     passwordElement.submit();
 
     driver.get(getPulseURL() + "clusterDetail.html");
-    WebElement userNameOnPulsePage =
-        (new WebDriverWait(driver, 30)).until(new ExpectedCondition<WebElement>() {
+    WebElement userNameOnPulsePage;
```
Comment (mcmellawatt) [inline geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/rules/WebDriverRule.java]: Why declare and assign on two different lines?

File: geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/Server.java [hunk 9]
```diff
@@ -54,11 +58,64 @@
...
+      Map<String, Object> env = new HashMap<String, Object>();
+
+      // set up Shiro Security Manager
+      Properties securityProperties = new Properties();
+      securityProperties.setProperty(TestSecurityManager.SECURITY_JSON, jsonAuthFile);
+      securityProperties.setProperty(SECURITY_MANAGER, TestSecurityManager.class.getName());
+
+      SecurityService securityService = SecurityServiceFactory.create(securityProperties);
+
+      // register the AccessControll bean
```
[the remark this answers is not in the document]
Comment (mhansonp) [reply geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/Server.java]: good feedback.

File: geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/Server.java [hunk 10]
```diff
@@ -54,11 +58,64 @@
...
+      AccessControlMBean acc = new AccessControlMBean(securityService);
+      ObjectName accessControlMBeanON = new ObjectName(ResourceConstants.OBJECT_NAME_ACCESSCONTROL);
+      MBeanServer platformMBeanServer = ManagementFactory.getPlatformMBeanServer();
+      platformMBeanServer.unregisterMBean(accessControlMBeanON);
+
+      // set up the AccessControlMXBean
```
[the remark this answers is not in the document]
Comment (mhansonp) [reply geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/Server.java]: good feedback.

File: geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/Server.java [hunk 11]
```diff
@@ -54,11 +58,64 @@
...
+
+    }
+    cs = null;
+
```
[the remark this answers is not in the document]
Comment (mhansonp) [reply geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/Server.java]: good feedback. removed.

File: geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/Server.java [hunk 12]
```diff
@@ -54,11 +58,64 @@
...
+      // register the AccessControll bean
+      AccessControlMBean acc = new AccessControlMBean(securityService);
+      ObjectName accessControlMBeanON = new ObjectName(ResourceConstants.OBJECT_NAME_ACCESSCONTROL);
+      MBeanServer platformMBeanServer = ManagementFactory.getPlatformMBeanServer();
+      platformMBeanServer.unregisterMBean(accessControlMBeanON);
+
+      // set up the AccessControlMXBean
+
+    }
+    cs = null;
```
[the remark this answers is not in the document]
Comment (mhansonp) [reply geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/Server.java]: good feedback. removed.

File: geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/Server.java (hunk 1 above)
Comment (mhansonp) [reply geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/Server.java]: good feedback. It was based on old code. I have removed all of that.

File: geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/Server.java (hunk 2 above)
Comment (mhansonp) [reply geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/Server.java]: same as above

File: geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/rules/ServerRule.java (hunk 4 above)
Comment (mhansonp) [reply geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/rules/ServerRule.java]: removed

File: geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/rules/ServerRule.java (hunk 5 above)
Comment (mhansonp) [reply geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/rules/ServerRule.java]: cleaned out.

File: geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/rules/ServerRule.java (hunk 6 above)
Comment (mhansonp) [reply geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/rules/ServerRule.java]: cleaned out.

File: geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/rules/ServerRule.java (hunk 7 above)
Comment (mhansonp) [reply geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/rules/ServerRule.java]: cleaned out.

File: geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/rules/WebDriverRule.java (hunk 8 above)
Comment (mhansonp) [reply geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/rules/WebDriverRule.java]: cleaned out.

**Pull request closed without merging**